src/main/java/uk/gov/hmcts/ccd/domain/model/common/HttpError.java

```diff
@@ -23,29 +25,63 @@
     private List<String> callbackErrors;
     private List<String> callbackWarnings;
 
-    public HttpError(Exception exception, HttpServletRequest request) {
-        final ResponseStatus responseStatus = exception.getClass().getAnnotation(ResponseStatus.class);
+    private final transient CatalogueResponse catalogueResponse;
+
+
+    public HttpError(final Exception exception, final HttpServletRequest request) {
+
+        this(exception, request, null, null);
+    }
+
+    public HttpError(final ApiException exception, final HttpServletRequest request) {
 
+        this(exception, request, exception.getCatalogueResponse(),null);
+    }
+
+    public HttpError(
+        final Exception exception,
+        final HttpServletRequest request,
+        final CatalogueResponse catalogueResponse,
+        final Integer status) {
+
+        final ResponseStatus responseStatus = exception.getClass().getAnnotation(ResponseStatus.class);
         this.exception = exception.getClass().getName();
         this.timestamp = LocalDateTime.now(ZoneOffset.UTC);
-        this.status = getStatus(responseStatus);
-        this.error = getErrorReason(responseStatus);
+        this.status = getStatusValue(status, responseStatus);
+        this.error = getErrorReason(status, responseStatus);
         this.message = exception.getMessage();
         this.path = request.getRequestURI();
+        this.catalogueResponse = catalogueResponse;
     }
 
-    private Integer getStatus(ResponseStatus responseStatus) {
-        if (null != responseStatus) {
+    private Integer getStatusValue(final Integer status, final ResponseStatus responseStatus) {
+        // Option 1: use status as it may be an override value for responseStatus
+        if (status != null) {
+            return status;
+
+        // Option 2: use responseStatus
+        } else if (null != responseStatus) {
             final HttpStatus httpStatus = getHttpStatus(responseStatus);
             if (null != httpStatus) {
                 return httpStatus.value();
             }
         }
 
+        // Option 3: fallback to use default
         return DEFAULT_STATUS;
     }
 
-    private HttpStatus getHttpStatus(ResponseStatus responseStatus) {
+    private HttpStatus getHttpStatus(final Integer status) {
+        // NB: don't return value if matches default
+        if (HttpStatus.INTERNAL_SERVER_ERROR.value() != status.intValue()) {
+            return HttpStatus.resolve(status.intValue());
+        }
+
+        return null;
+    }
```

Comment:
The logic here doesn't look consistent, as it is returning a null for 500. Either the code should be fixed or the method be renamed to imply the logic inside it.